pkg/client/core/service_container_repository_client.go

```diff
@@ -0,0 +1,126 @@
+package core
+
+import (
+	"context"
+
+	"github.com/oslokommune/okctl/pkg/apis/okctl.io/v1alpha1"
+
+	"github.com/oslokommune/okctl/pkg/cfn"
+
+	"github.com/oslokommune/okctl/pkg/client/store"
+
+	"github.com/oslokommune/okctl/pkg/api"
+	"github.com/oslokommune/okctl/pkg/client"
+	"github.com/oslokommune/okctl/pkg/spinner"
+)
+
+type containerRepositoryService struct {
+	spinner spinner.Spinner
+	api     client.ContainerRepositoryAPI
+	store   client.ContainerRepositoryStore
+	state   client.ContainerRepositoryState
+	report  client.ContainerRepositoryReport
+
+	provider v1alpha1.CloudProvider
+}
+
+// CreateContainerRepository handles api, state, store and report orchistration for creation of a container repository
+func (c *containerRepositoryService) CreateContainerRepository(_ context.Context, opts client.CreateContainerRepositoryOpts) (*client.ContainerRepository, error) {
+	err := c.spinner.Start("container registry")
```

Comment:
You can keep it if you want to, but I will remove it in my state PR, so if you waaaant to 😬 (the spinner)